src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ScannerZipInstaller.java

```diff
@@ -82,16 +96,32 @@ public File installOrUpdateScanner() throws BlackDuckIntegrationException {
         File scannerExpansionDirectory = new File(installDirectory, ScannerZipInstaller.BLACK_DUCK_SIGNATURE_SCANNER_INSTALL_DIRECTORY);
         scannerExpansionDirectory.mkdirs();
 
-        File versionFile = null;
-        try {
-            versionFile = retrieveVersionFile(scannerExpansionDirectory);
-        } catch (IOException e) {
-            throw new BlackDuckIntegrationException("Trying to install the scanner but could not create the version file: " + e.getMessage(), e);
-        }
-
+        File versionFile = new File(scannerExpansionDirectory, ScannerZipInstaller.VERSION_FILENAME);
         HttpUrl downloadUrl = getDownloadUrl();
+
         try {
-            downloadIfModified(scannerExpansionDirectory, versionFile, downloadUrl);
+            String connectedBlackDuckVersion = StringUtils.trim(blackDuckRegistrationService.getBlackDuckServerData().getVersion());
+
+            if (!versionFile.exists()) {
+                logger.info("No version file exists, assuming this is new installation and the signature scanner should be downloaded.");
+                downloadSignatureScanner(scannerExpansionDirectory, downloadUrl);
+                // Version file creation should happen after successful download
+                logger.debug("The version file has not been created yet so creating it now.");
+                FileUtils.writeStringToFile(versionFile, connectedBlackDuckVersion, Charset.defaultCharset());
+                return installDirectory;
+            }
+
+            // A version file exists, so we have to compare to determine if a download should occur.
+            String localScannerVersion = FileUtils.readFileToString(versionFile, Charset.defaultCharset());
+            logger.debug(String.format("Last version downloaded: %s", localScannerVersion));
+
+            if (!localScannerVersion.equals(connectedBlackDuckVersion)) {
+                logger.info(String.format("The signature scanner should be downloaded. Current version: %s Black Duck version: %s", localScannerVersion, connectedBlackDuckVersion));
```

Comment:
I'd maybe change this to say "The signature scanner should be downloaded. Local Signature Scanner version is %s, but Black Duck version is %s"